jerry-core/ecma/builtin-objects/ecma-builtin-array.inc.h

```diff
@@ -19,6 +19,8 @@
 
 #include "ecma-builtin-helpers-macro-defines.inc.h"
 
+#ifndef CONFIG_DISABLE_ARRAY_BUILTIN
+
```

Comment:
Most of the time, I agree with you. Therefore, I've made an experiment to build up the inclusion graph of all processed files and figure out how the inclusions are guarded. After getting the work done, I found that somehow the tool has not added the builtin guards for the `.inc.h` files, contrary to expectations.

It turns out that the reason is that the builtin `.inc.h`s are transitively and indirectly included. The source files define `BUILTIN_INC_HEADER_NAME` and the include `ecma-builtin-internal-routines-template.inc.h`, which in turn includes `BUILTIN_INC_HEADER_NAME` (which gets expanded by the preprocessor). I couldn't find a way to track such indirect inclusions in an included file.

I've reverted that experiment.